#! /bin/sh /usr/share/dpatch/dpatch-run
## DP: Fixes:
## DP: http://redmine.ruby-lang.org/issues/show/1392
## DP: Object#extend leaks memory on Ruby 1.9.1
## DP: 
## DP: https://bugs.launchpad.net/ubuntu/+source/ruby1.9.1/+bug/514322
## DP: Ruby 1.9.1 leaks memory when using Module.include
## DP: 
## DP: https://bugs.launchpad.net/ubuntu/+source/ruby1.9.1/+bug/529011
## DP: Ruby 1.9.1 leaks memory in Lucid on AMD64
## DP: 
## DP: commit f8d96940549f573c92c1d1aa74270ac4099871d8
## DP: Author: wanabe <wanabe@b2dd03c8-39d4-4d8f-98ff-823fe69b080e>
## DP: Date:   Sun Jan 31 04:33:33 2010 +0000
## DP:
## DP:    * gc.c (obj_free): free rb_classext_t of eigenclass.  [Bug #1392]
## DP:    
## DP:    
## DP:    git-svn-id: http://svn.ruby-lang.org/repos/ruby/trunk@26515 b2dd03c8-39d4-4d8f-98ff-823fe69b080e
## DP:
@DPATCH@
Index: b/gc.c
===================================================================
--- a/gc.c	2010-07-20 21:45:44.000000000 +0900
+++ b/gc.c	2010-07-20 21:46:19.000000000 +0900
@@ -1905,6 +1905,7 @@
 	break;
       case T_ICLASS:
 	/* iClass shares table with the module */
+	xfree(RANY(obj)->as.klass.ptr);
 	break;
 
       case T_FLOAT:
